src/server/scripts/Outland/HellfireCitadel/MagtheridonsLair/boss_magtheridon.cpp

```diff
@@ -16,617 +15,590 @@
+                }
             }
-            void MoveInLineOfSight(Unit* who) override
 
+            void JustDied(Unit* /*killer*/) override
             {
-                if (!trigger)
-                    ScriptedAI::MoveInLineOfSight(who);
+                Talk(SAY_DEATH);
+                _JustDied();
+                instance->SetData(DATA_MANTICRON_CUBE, ACTION_DISABLE);
+            }
+
+            void KilledUnit(Unit* who) override
+            {
+                if (who->GetTypeId() == TYPEID_PLAYER)
+                    Talk(SAY_SLAY);
             }
 
             void UpdateAI(uint32 diff) override
             {
-                if (trigger)
+
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                events.Update(diff);
+
+                while (uint32 eventId = events.ExecuteEvent())
                 {
-                    if (trigger == 1)
+                    switch (eventId)
                     {
-                        if (FireBlast_Timer <= diff)
-                        {
-                            DoCast(me, SPELL_DEBRIS_DAMAGE, true);
-                            trigger = 3;
-                        }
-                        else FireBlast_Timer -= diff;
+                        case EVENT_BERSERK:
+                            DoCastSelf(SPELL_BERSERK);
+                            break;
+                        case EVENT_CLEAVE:
+                            DoCastVictim(SPELL_CLEAVE);
+                            events.Repeat(Seconds(10));
+                            break;
+                        case EVENT_BLAZE:
+                            me->CastCustomSpell(SPELL_BLAZE_TARGET, SPELLVALUE_MAX_TARGETS, 1);
+                            events.Repeat(Seconds(20));
+                            break;
+                        case EVENT_QUAKE:
+                            me->CastCustomSpell(SPELL_QUAKE, SPELLVALUE_MAX_TARGETS, 5);
+                            events.Repeat(Seconds(60));
+                            break;
+                        case EVENT_START_FIGHT:
+                            CombatStart();
+                            break;
+                        case EVENT_RELEASED:
+                            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_NOT_SELECTABLE);
+                            me->SetInCombatWithZone();
+                            instance->SetData(DATA_MANTICRON_CUBE, ACTION_ENABLE);
+                            events.ScheduleEvent(EVENT_CLEAVE, Seconds(10));
+                            events.ScheduleEvent(EVENT_BLAST_NOVA, Seconds(60));
+                            events.ScheduleEvent(EVENT_BLAZE, Seconds(20));
+                            events.ScheduleEvent(EVENT_QUAKE, Seconds(35));
+                            events.ScheduleEvent(EVENT_BERSERK, Minutes(20));
+                            events.CancelEvent(EVENT_TAUNT);
```

Comment:
fixed in 8fbc8b5e1a5145163def8fefafbb41b1b0533e42 xd
change code while play starcraft is not a good idea 💃